src/auth/authService.ts

```diff
@@ -77,7 +77,15 @@ export class AuthService {
 
   async verifyEmail(code: number, user_id: number){
     let checkCode = await this.repository.getCode(user_id);
-    if(checkCode !== code){throw new WrongCode()}
+
+    if(checkCode.code !== code){throw new WrongCode()}
+    else if ( checkCode.expires_at < Date.now()){
+      const newCode = this.generateCode();
+      const user = await this.repository.findUserById(user_id)
+      await this.repository.addEmailCode(newCode, user_id);
+      await this.mail.sendMail(user_id, user.email, code);
+      return "expired";
```

Comment:
maybe throw custom error
and catch it in try {} catch {} statement.
also need to send 400 status code